src/state/actions/recipes.js

```diff
@@ -0,0 +1,48 @@
+import * as actionTypes from './actionTypes';
+import { fetchRecipes } from '../../api/fetch';
+import { APP_ID, APP_KEY } from '../../api/passwords';
+
+export const saveRecipes = (ingredients, kcal) => {
+  return dispatch => {
+    dispatch(saveRecipesStart());
+    const kcalFilled = (kcal.length > 0 ) && `&calories=${kcal}`;
+    const url = `https://api.edamam.com/search?app_id=${APP_ID}&app_key=${APP_KEY}&q=${ingredients}&${kcalFilled}`;
```

Comment:
Wrong url, you have `&${kcalFilled}` kcalFilled can be undefined or string with `&`, so result url will look like `q=ingredient1&&10` (should be 1 amp)